src/main/java/uk/ac/cam/cl/dtg/isaac/api/GameboardsFacade.java

```diff
@@ -323,6 +335,71 @@ public final Response getFastTrackConceptProgress(@Context final Request request
         }
     }
 
+    /**
+     * REST endpoint to retrieve every question (and its status) associated with a particular FastTrack gameboard and
+     * history.
+     *
+     * @param request usually used for caching.
+     * @param httpServletRequest so that we can extract the users session information if available.
+     * @param gameboardId the unique id of the FastTrack gameboard which links the questions.
+     * @param upper the latest upper level question that is in the history.
+     * @param currentConceptTitle the concept title that the user is currently working on.
+     * @return a Response containing a list of augmented gameboard items for the gamebaord-concept pair or an error.
+     */
+    @GET
+    @Path("fasttrack/{gameboard_id}/concepts_with_upper/")
```

Comment:
We should consolidate the two fasttrack concept endpoints, only do the upper search if upper is also included in the request